From 81ddfd1ba417c475b5ae843efc360f7a49bd4b7a Mon Sep 17 00:00:00 2001
From: oliverwen <oliverwen@google.com>
Date: Thu, 30 Jun 2016 09:01:05 -0700
Subject: [PATCH] CHROMIUM: smart antenna algorithm: bugfix

 - fix the bug of station counter per antenna

TEST=Pass compilation and building of ChromeOS image.
Disconnect and reconnect station with AP while smart antenna
training is in progress. nsta_sel_ant should track the number of
stations per antenna correctly, and always be less or equal to
num_sta_conneted,i.e., the total number of stations associated with AP.

BUG=chrome-os-partner:51964
CQ-DEPEND=CL:341552

Signed-off-by: oliverwen <oliverwen@google.com>
(cherry picked from commit 6748644a114e684276796c8e37e5ea87e9becac0)
(source: wireless-3.18)

Change-Id: Id12b83c1792960af8a042c74615de9ef1a5d26ae
Reviewed-on: https://chromium-review.googlesource.com/341554
Commit-Ready: Yixiang Li <yixiang@google.com>
Tested-by: Oliver Wen <oliverwen@chromium.org>
Reviewed-by: Yixiang Li <yixiang@google.com>
---
 drivers/net/wireless/ath/ath10k/smart_ant.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath10k/smart_ant.c b/drivers/net/wireless/ath/ath10k/smart_ant.c
index 925fe5fd39d..6d1db38d319 100644
--- a/drivers/net/wireless/ath/ath10k/smart_ant.c
+++ b/drivers/net/wireless/ath/ath10k/smart_ant.c
@@ -1534,7 +1534,7 @@ void ath10k_smart_ant_sta_disconnect(struct ath10k *ar,
 	spin_unlock_bh(&sa_sta->cfg_lock);
 
 	info->num_sta_conneted--;
-	info->num_sta_per_ant[sa_sta->train_info.sel_ant]--;
+	info->num_sta_per_ant[sa_sta->train_info.prev_sel_ant]--;
 
 	arsta->smart_ant_sta = NULL;
 	kfree(arsta->smart_ant_sta);
-- 
2.43.0

